src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    @Test
+    public void should_update_project() throws Exception {
+        Mockito.when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        Mockito.when(projectRepository.findById(ArgumentMatchers.any(Long.class))).thenReturn(Optional.of(project));
+        Mockito.when(projectRepository.save(ArgumentMatchers.any(Project.class))).thenReturn(project);
+        Assert.assertTrue(projectService.updateProject(userPrincipal, project.getId(), projectRequest).getName().contains(project.getName()));
+        Assert.assertTrue(projectService.updateProject(userPrincipal, project.getId(), projectRequest).getTitle().contains(project.getTitle()));
+        Assert.assertTrue(projectService.updateProject(userPrincipal, project.getId(), projectRequest).getDescription().contains(project.getDescription()));
+    }
+
+    @Test
+    public void should_delete_project() throws Exception {
+        Mockito.when(projectRepository.findById(ArgumentMatchers.any(Long.class))).thenReturn(Optional.of(project));
+        Mockito.when(userDao.findByPrivateId(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        Assert.assertEquals(projectService.deleteProject(userPrincipal, project.getId()).getStatusCodeValue(), HTTPResponse.SC_OK);
+    }
+
+    @Test
+    public void should_find_by_Name() throws Exception {
+        Mockito.when(projectRepository.findByName(ArgumentMatchers.any(String.class))).thenReturn(Optional.of(project));
+        Mockito.when(userDao.findByPublicId(ArgumentMatchers.any(Long.class))).thenReturn(Optional.of(user));
+        Assert.assertTrue(projectService.findByName(userPrincipal, project.getName()).getDescription().contains(project.getDescription()));
+        Assert.assertTrue(projectService.findByName(userPrincipal, project.getName()).getBody().contains(project.getBody()));
+        Assert.assertTrue(projectService.findByName(userPrincipal, project.getName()).getName().contains(project.getName()));
+    }
+
+    @Test
+    public void should_find_by_technology() throws Exception {
+        Mockito.when(technologyRepository.findById(ArgumentMatchers.any(Long.class))).thenReturn(Optional.of(technology));
+        Mockito.when(projectRepository.findByIdIn(ArgumentMatchers.anyList(), ArgumentMatchers.isA(Pageable.class))).thenReturn(page);
+        Mockito.when(userDao.findByPublicId(ArgumentMatchers.any(Long.class))).thenReturn(Optional.of(user));
+
+        Assert.assertTrue(projectService.findByTechnology(userPrincipal, technology.getId(), 0, 10).getContent().get(0).getName().contains(project.getName()));
+        Assert.assertTrue(projectService.findByTechnology(userPrincipal, technology.getId(), 0, 10).getContent().get(0).getBody().contains(project.getBody()));
+        Assert.assertTrue(projectService.findByTechnology(userPrincipal, technology.getId(), 0, 10).getContent().get(0).getDescription().contains(project.getDescription()));
+    }
```

Comment:
1. Mockito i Assert to są metody statyczne, nie piszesz Mockito.when tylko samo when. Assert to samo
2.Given When Then dla przejrzystego kodu testów
3. Po co przy każdej assercji wywoływać metodę. Zamiast w jednym treści wywołać raz metodę wywołujesz ja po kilka razy. To nie jest optymalne. Wywołuje się ją raz, zapisuje w zmiennej a potem się sprawdza wartości. 
4.@DisplayName()
5.@TestMethodOrder(MethodOrderer.Alphanumeric.class)
6.verify metoda w testach